lib/sidekiq/redis_connection.rb

```diff
@@ -7,10 +7,7 @@ class RedisConnection
     class << self
 
       def create(options={})
-        url = options[:url] || determine_redis_provider
-        if url
-          options[:url] = url
-        end
+        options[:url] ||= determine_redis_provider
```

Comment:
@mperham I find [this](https://github.com/redis/redis-rb/blob/master/lib/redis/client.rb#L384). You can see if `options[:url]` equal `nil` redis connect to `defaults[:url]`. This shows that my changes are correct :tada: